### PR TITLE
Update build script and targeting nugets to allow the project to build on the windows-2022 image

### LINKED ISSUE #77: The build agent needs an upgrade to build against the new ProjFS API
**Context**
The build agent used in the PR Build/Test pipeline runs Windows Server 10.0.17763. There are some ProjFS APIs that were introduced in later Windows versions, such as 2004. Notably, the symlink API (https://docs.microsoft.com/en-us/windows/win32/api/projectedfslib/ns-projectedfslib-prj_extended_info - 10.0; Build 19041) that was used in  https://github.com/microsoft/ProjFS-Managed-API/pull/73.

One approach to solving this could be using the `windows-2022` [image](https://github.com/actions/virtual-environments/blob/win22/20211122.0/images/win/Windows2022-Readme.md). It seems as if this would also need changes in the build scripts to allow builds on that environment (for example, .NET Framework 4.6.1 SDK is missing there, and the VS version is 2022 and not 2019 that current script expects). Christian and I tried a couple of minor changes, but it seems as if this requires additional investigation.

Please let me know if any additional details are needed here - I will be happy to help!

### PR DESCRIPTION
**Pull request by yerudako**
Fixes #77. This change is combined from @cgallred's and mine changes and based on[ this run](https://dev.azure.com/projfs/ci/_build/results?buildId=168&view=logs&j=cfa20e98-6997-523c-4233-f0a7302c929f) Christian triggered with them, it might be enough changes to use Windows 2022 (Please ignore the test failures - these are related to the symlink branch and are not present in the baseline)

Changelog:
 - Props file updated to reference correct Windows SDK versions
 - Targeting nugets used to build 4.6.1 on windows-2022 env (which only has .NET Core and .NET 4.8 present)
 - Redundant components removed from the vswhere check
 - Version limits updated in the vswhere check

### CODE REVIEW

**Comment by narasamdya** [comment]
Why do we need .NET 4.6.1 ? Is .NET 4.8 not sufficient?

**Comment by yerudako** [comment]
> Why do we need .NET 4.6.1 ? Is .NET 4.8 not sufficient?

@narasamdya the project builds in 4.6.1, so I assumed that it's supposed to be used by older client apps running 4.6.1 and added the targeting pack rather than retargeting it to 4.8. I would defer to @cgallred for whether it is critical to support 4.6.1 or whether we should migrate to 4.8.

**Comment by cgallred** [comment]
> > Why do we need .NET 4.6.1 ? Is .NET 4.8 not sufficient?
> 
> @narasamdya the project builds in 4.6.1, so I assumed that it's supposed to be used by older client apps running 4.6.1 and added the targeting pack rather than retargeting it to 4.8. I would defer to @cgallred for whether it is critical to support 4.6.1 or whether we should migrate to 4.8.

It targets 4.6.1 because for a long time there had been a requirement that it needed to run on RS1 (see [this PR comment](https://github.com/microsoft/ProjFS-Managed-API/pull/40#issuecomment-566729386)). That requirement no longer exists. From what I understand of the platforms .NET 4.8 is supported on (based on [this doc](https://docs.microsoft.com/en-US/lifecycle/faq/dotnet-framework#what-is-the-lifecycle-policy-for-different-versions-of--net-framework-)), 4.8 is the current Framework version on all supported OS versions, right? Plus 4.6.1 will be retired on April 26th. 

I think we can update to 4.8.

File: scripts/BuildProjFS-Managed.bat [hunk 1]
```diff
@@ -45,6 +45,7 @@ call "%VsDir%\VC\Auxiliary\Build\vcvarsall.bat" x86_amd64
  
 :: Restore all dependencies and run the build.
 pushd "%PROJFS_SRCDIR%"
+%nuget% install Microsoft.NETFramework.ReferenceAssemblies.net461  -Version 1.0.2 -OutputDirectory %PROJFS_PACKAGESDIR%
```
Comment (cgallred) [inline scripts/BuildProjFS-Managed.bat:48]: This doesn't need to be here anymore does it?

**Review by cgallred — CHANGES_REQUESTED** [review]

File: scripts/BuildProjFS-Managed.bat (hunk 1 above)
Comment (yerudako) [reply scripts/BuildProjFS-Managed.bat]: Absolutely. Thanks for catching this!

**Review by cgallred — APPROVED** [review]

**Comment by yerudako** [comment]
@cgallred thanks for reviewing! Can I also ask you to merge it - it says I am not authorized to merge the Pull Request?

**Pull request merged**